worker/src/worker.rs

```
@@ -0,0 +1,86 @@
+            _page_type_marker: PhantomData,
+            _data_type_marker: PhantomData,
+        }
+    }
+
+    /// Starts the worker. It will now listen to the manager for new tasks are resolve those.
+    /// This is a blocking operation.
+    pub fn start(&self) {
+        self.manager.start_listening(|task| {
+            println!("Worker {} received task {}", self.name, task.url);
+            // TODO: Proper error handling
+            match self.downloader.fetch_page(task) {
+                Err(e) => {
+                    eprintln!("{} failed to download a page.", self.name);
+                    TaskProcessResult::Err
+                }
+                Ok(page) => {
+                    match self.extractor.extract_content(page) {
+                        Err(e) => {
+                            eprintln!("{} failed to extract data from page.", self.name);
+                            TaskProcessResult::Err
+                        }
+                        Ok((tasks, data)) => {
+                            for datum in data {
+                                if let Err(e) = self.archive.archive_content(datum) {
+                                    eprintln!("{} failed archiving some data.", self.name);
+                                }
+                            }
+
+                            for task in &tasks {
+                                if let Ok(exists) = self.manager.contains(task) {
+                                    if !exists {
+                                        if let Err(_) = self.manager.submit_task(task) {
+                                            eprintln!("{} failed submitting a new task to the manager.", self.name);
+                                        }
+                                    }
+                                } else {
+                                    eprintln!("{} failed to check if a new task is present in the collection. Skipping that task.", self.name)
+                                }
```

Again why are we failing silently?

---

See my other comment.